src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -7612,6 +7597,88 @@ interface CorporationInfo {
   divisions: Division[];
 }
 
+/**
+ * Corporation related constants
+ * @public
+ */
+interface CorpConstants {
+  /** Name of the corporation */
+  states: string[];
+  /** Funds available */
+  unlocks: string[];
+  /** Revenue per second this cycle */
+  upgrades: string[];
+  /** Expenses per second this cycle */
+  researches: Record<string, string[]>;
+  /** Indicating if the company is public */
+  jobs: string[];
+  /** Total number of shares issues by this corporation */
+  bribeToRepRatio: number;
+  /** Amount of share owned */
+  baseMaxProducts: number;
+  /** Cooldown until shares can be sold again */
+  cityExpandCost: number;
+  /** Amount of acquirable shares. */
+  warehousePurchaseCost: number;
+  /** Price of the shares */
+  coffeeCost: number;
+  /** Fraction of profits issued as dividends */
+  materials: Record<string, materialInfo>;
+  /** Tax applied on your earnings as a shareholder */
+  products: Record<string, productInfo>;
+  /** Your earnings as a shareholder per second this cycle */
+  divisions: Record<string, divisionInfo>;
```

Comment:
Record<string, X> is a bad type to provide to the Player (and just in general). Even though it's a fixed object with specific keys, Record<string, X> doesn't know what keys it has and doesn't provide autocompletion assistance to the user, and doesn't know when an invalid key was provided.

I don't mind tweaking these types myself, but it will probably wreck your other gameplay PR with a bunch of merge conflicts if I merge this one and then mess around with a bunch of it afterwards. So I'd rather merge them both together so my subsequent changes won't cause an issue. Will probably just wait until after 2.2.0 full release to merge these.

Reply:
To be honest I don't know how to make those better :/

Reply:
No worries, I don't mind tweaking types myself, I just don't want to make it hell to rebase your other major PR that touches all the same stuff. Is the gameplay changes PR ready ish? If so I'll just merge both and mess with types and whatnot afterwards.

Reply:
yeah, they are merge-ready as far as I'm concerned

Reply:
Wait where's the gameplay branch though? Did you submit a PR for that one?

Reply:
I removed it, the corp rework stuff need some work before being pushed. Maybe it'll be good enough for 2.4 or 2.3.1 :)